core/src/test/java/org/apache/accumulo/core/data/NamespaceIdTest.java

```diff
@@ -14,57 +14,63 @@
-    Long initialSize = Namespace.ID.cache.asMap().entrySet().stream().count();
-    Namespace.ID nsId = Namespace.ID.of(namespaceString);
-    assertEquals(initialSize + 1, Namespace.ID.cache.asMap().entrySet().stream().count());
-    assertEquals(namespaceString, nsId.canonicalID());
+    Long initialSize = NamespaceId.cache.asMap().entrySet().stream().count();
+    NamespaceId nsId = NamespaceId.of(namespaceString);
+    assertEquals(initialSize + 1, NamespaceId.cache.asMap().entrySet().stream().count());
+    assertEquals(namespaceString, nsId.canonical());
   }
 
   @Test
   public void testCacheNoDuplicates() {
+    // the next line just preloads the built-ins, since they now exist in a separate class from
+    // NamespaceId, and aren't preloaded when the NamespaceId class is referenced
+    assertNotSame(Namespace.ACCUMULO.id(), Namespace.DEFAULT.id());
+
     String namespaceString = "namespace-" + name.getMethodName();
-    Long initialSize = Namespace.ID.cache.asMap().entrySet().stream().count();
-    Namespace.ID nsId = Namespace.ID.of(namespaceString);
-    assertEquals(initialSize + 1, Namespace.ID.cache.asMap().entrySet().stream().count());
-    assertEquals(namespaceString, nsId.canonicalID());
+    Long initialSize = NamespaceId.cache.asMap().entrySet().stream().count();
+    NamespaceId nsId = NamespaceId.of(namespaceString);
+    assertEquals(initialSize + 1, NamespaceId.cache.asMap().entrySet().stream().count());
+    assertEquals(namespaceString, nsId.canonical());
 
     // ensure duplicates are not created
-    Namespace.ID builtInNamespaceId = Namespace.ID.of("+accumulo");
-    assertSame(Namespace.ID.ACCUMULO, builtInNamespaceId);
-    builtInNamespaceId = Namespace.ID.of("+default");
-    assertSame(Namespace.ID.DEFAULT, builtInNamespaceId);
-    nsId = Namespace.ID.of(namespaceString);
-    assertEquals(initialSize + 1, Namespace.ID.cache.asMap().entrySet().stream().count());
-    assertEquals(namespaceString, nsId.canonicalID());
-    Namespace.ID nsId2 = Namespace.ID.of(namespaceString);
-    assertEquals(initialSize + 1, Namespace.ID.cache.asMap().entrySet().stream().count());
+    NamespaceId builtInNamespaceId = NamespaceId.of("+accumulo");
+    assertSame(Namespace.ACCUMULO.id(), builtInNamespaceId);
+    builtInNamespaceId = NamespaceId.of("+default");
+    assertSame(Namespace.DEFAULT.id(), builtInNamespaceId);
+    nsId = NamespaceId.of(namespaceString);
+    assertEquals(initialSize + 1, NamespaceId.cache.asMap().entrySet().stream().count());
+    assertEquals(namespaceString, nsId.canonical());
+    NamespaceId nsId2 = NamespaceId.of(namespaceString);
+    assertEquals(initialSize + 1, NamespaceId.cache.asMap().entrySet().stream().count());
     assertSame(nsId, nsId2);
   }
 
   @Test(timeout = 60_000)
   public void testCacheDecreasesAfterGC() {
-    Long initialSize = Namespace.ID.cache.asMap().entrySet().stream().count();
+    Long initialSize = NamespaceId.cache.asMap().entrySet().stream().count();
```

Comment:
I have been working on getting Accumulo to build with Java 11.  The `testCacheDecreasesAfterGC` test is failing (due to timeout) for `NamespaceTest` and the `TableTest` (in master before these changes). I am not sure why. It might be nice to try to run these changes with Java 11.

Reply:
These changes aren't substantial. I doubt it would impact that test.

Reply:
I was just pointing out that this test is failing with Java 11 if you wanted to fix that now.  This can always be fixed later.

Reply:
Will probably have to wait until later. I don't have Java 11 set up anywhere yet.